### ISSUE TITLE
A failure of reproduction in AIR example

### ISSUE THREAD

**Issue by Minju-Jung** [uber/pyro#637]
I tried to reproduce the result of AIR reported in http://pyro.ai/examples/air.html.
I used the default setting of main.py as follows:
`python main.py --progress-every 100 --eval-every 100 --viz --cuda`

However, the count accuracy did not increased more than 33%.
In case of zero digit, the network successfully estimated the number of digits as zero, but the network wrongly estimated the number of digits as three in the other cases (one and two digits).

Could somebody help me to solve the problem?

> i=10000, epochs=10.67, elapsed=0.11, elbo=-622.77
i=10000, accuracy=0.19103333333333333, counts=[[11430, 5505, 2139, 938], [152, 32, 15, 19858], [2, 0, 0, 19929]]
i=20000, epochs=21.33, elapsed=0.22, elbo=-630.32
i=20000, accuracy=0.26921666666666666, counts=[[16150, 3750, 112, 0], [64, 3, 1, 19989], [0, 0, 0, 19931]]
i=30000, epochs=32.00, elapsed=0.33, elbo=-643.42
i=30000, accuracy=0.32221666666666665, counts=[[19331, 681, 0, 0], [86, 2, 0, 19969], [0, 0, 0, 19931]]
i=40000, epochs=42.67, elapsed=0.44, elbo=-641.45
i=40000, accuracy=0.32425, counts=[[19455, 557, 0, 0], [44, 0, 0, 20013], [0, 0, 0, 19931]]

**Comment by martinjankowiak** [comment]
have you tried the settings explicitly listed at the end of the tutorial?

**Comment by Minju-Jung** [comment]
@martinjankowiak Could you kindly let me know what setting do I need to follow?
I added the option --z-pres-prior 0.01, but it looks the same as before.

**Comment by martinjankowiak** [comment]
refer to the bottom of the tutorial.

**Comment by Minju-Jung** [comment]
@martinjankowiak I checked the tutorial as you recommended, but still I cannot solve the problem.

1. Inference is unlikely to recover correct object counts unless a small prior success probability for z_pres is used. In [1] this probability was annealed from a value close to one to 1e-5 (or less) during optimization, though we found that a fixed value of around 0.01 worked well with our implementation.
--> I added the option --z-pres-prior 0.01.

2. We initialize the decoder network to generate mostly empty objects initially. (Using the --decoder-output-bias argument.) This encourages the guide to explore the use of objects to explain the input early in optimization. Without this each object is a mid-gray square which is heavily penalized by the likelihood, prompting the guide to turn most steps off.
--> There is no exact setting for --decoder-output-bias. I set --decoder-output-bias to -1.

3. It is reported to be useful in practice to use a different learning rate for the baseline network. This is straight forward to implement in Pyro by tagging modules associated with the baseline network and passing multiple learning rates to the optimizer. (See the section on optimizers in part I of the SVI tutorial for more detail.) In [1] a learning rate of 1e-4 was used for the guide network, and a learning rate of 1e-3 was used for the baseline network. We found it necessary to use a larger learning rate for the baseline network in order to make progress on count accuracy at a similar rate to [1]. This difference is likely caused by Pyro setting up a slightly different baseline loss.
--> This are the default setting of main.py
